### DISCUSSION TITLE
possibility of defining a repo and a branch for each project individually

### DISCUSSION THREAD

**Discussion by samar-elsayed** [apache/camel-karavan#1119] [Q&A] [answered]
Hi,

is it possible to have a way to define a repo for each project ? and a branch to push the files to?

**Comment by arheom** [comment]
From what I can see inside the application.properties: https://github.com/apache/camel-karavan/blob/main/karavan-web/karavan-app/src/main/resources/application.properties:
```
karavan.git-repository=http://gitea:3000/karavan/karavan.git
karavan.git-username=karavan
karavan.git-password=karavan
karavan.git-branch=main
```
you could choose a branch, but all inside same repo. Each project would be a folder inside that repo, but you could eventually have different branches for different environments, for example.

**Comment by mgubaidullin** [comment, marked as answer]
Thanks @arheom 

@samar-elsayed you could configure only one repository for Karavan instance.

**Reply by samar-elsayed** [reply]
Thanks @arheom @mgubaidullin 
One last thing, is there an export option inside karavan UI like the one inside the VS code plugin?

**Reply by arheom** [reply]
I havent seen an export feature in the UI, but inside the templates, you can check your build script (for kubernetes: kubernetes-build.sh) and inside you have the export:
`jbang -Dcamel.jbang.version=${CAMEL_VERSION} camel@apache/camel export --local-kamelet-dir=${KAMELETS_DIR}`
you could update the script to do something with the export, like push it into the git / file share. Further on, the export gets as an input to mvn command, which you could also update as you need. So, when you press the build button, you could do the export and process it as you need it.

**Reply by mgubaidullin** [reply]
How build container works in Karavan
1. Export project folder (with yaml, properties, java, etc) to maven project using `jbang camel@apache/camel export`
2. execute `mvn package` that compiles code, package jar, build and push container image and apply deployment to k8s

There is no need to do export in UI when you use Karavan Application. 

However if yo want to leave Karavan for some reason, you could export your projects to maven projects and continue develop them traditional Maven way. So you are not locked-in to Karavan
